addon/routes/fd-appstruct-form.js

```diff
@@ -0,0 +1,64 @@
+import Ember from 'ember';
+import { Query } from 'ember-flexberry-data';
+const { Builder, FilterOperator } = Query;
+
+export default Ember.Route.extend({
+
+  currentProjectContext: Ember.inject.service('fd-current-project-context'),
+
+  model: function() {
+    let stagePk = this.get('currentProjectContext').getCurrentStage();
+    let _this = this;
+    let modelHash = {
+      implementations: undefined,
+      forms: undefined,
+      applications: undefined
+    };
+
+    return new Ember.RSVP.Promise(function (resolve) {
```

Comment:
This promise will never resolved if `promiseImplementation`, `promiseForms`, or `promiseApplication` will be rejected.